src/argilla/server/contexts/datasets.py

```diff
@@ -531,12 +574,15 @@ async def create_records(
         )
 
         if record_create.responses:
-            for response in record_create.responses:
-                # TODO(gabrielmbmb): the result of this query should be cached
-                if not await accounts.get_user_by_id(db, response.user_id):
-                    raise ValueError(f"Provided user_id: {response.user_id!r} is not a valid user id")
-
-                validate_response_values(dataset, values=response.values, status=response.status)
+            users_ids_cache: Set[UUID] = set()
+            for i, response in enumerate(record_create.responses):
+                try:
+                    users_ids_cache = await validate_user(db, response.user_id, users_ids_cache)
```

Comment:
Why do we need to validate that the response user exists?

I can understand a check to know if the user belongs to the dataset workspace (we're not doing that right?) but only for existence? (The foreign key is already doing that).

Reply:
Admins can provide multiple responses for a record (and each one from a different user). This validation tries to check that responses were properly configured without raising an undescribable `IntegrityError`